pandas/core/arrays/string_arrow.py

```diff
@@ -755,15 +756,31 @@ def _str_map(self, f, na_value=None, dtype: Dtype | None = None):
             return lib.map_infer_mask(arr, f, mask.view("uint8"))
 
     def _str_contains(self, pat, case=True, flags=0, na=np.nan, regex=True):
-        if not regex and case:
-            result = pc.match_substring(self._data, pat)
-            result = BooleanDtype().__from_arrow__(result)
-            if not isna(na):
-                result[isna(result)] = bool(na)
-            return result
-        else:
+        if flags:
             return super()._str_contains(pat, case, flags, na, regex)
 
+        if regex:
+            if hasattr(pc, "match_substring_regex") and case:
```

Comment:
Maybe it would be good to add a comment like "added in pyarrow x.x", so we can more easily clean up those `hasattr` checks if we later raise the minimum version

Reply:
yes. I was thinking the same to address https://github.com/pandas-dev/pandas/pull/41219#discussion_r623435346